src/main/java/prefs/Preferences.java

```diff
@@ -11,122 +17,262 @@
+        String jsonString = jsonHelper.toJsonString(config, configClass);
+        FileHelper fileHelper = new FileHelper(configDirectory, configFileName);
+        fileHelper.writeFileContents(jsonString);
     }
 
-    // Last login credentials. While the main UI is running (i.e. logged in successfully), last login
-    // credentials are guaranteed to be the current user's credentials thanks to setLastLoginCredentials
-    // being called immediately after a successful login in LoginDialog.
+    /**
+     * Last login credentials. While the main UI is running (i.e. logged in successfully), last login
+     * credentials are guaranteed to be the current user's credentials thanks to setLastLoginCredentials
+     * being called immediately after a successful login in LoginDialog.
+     */
     public String getLastLoginPassword() {
-        return global.getLastLoginPassword();
+        return sessionConfig.getLastLoginPassword();
     }
 
+    /**
+     * Retrieves the last login username
+     */
     public String getLastLoginUsername() {
-        return global.getLastLoginUsername();
+        return sessionConfig.getLastLoginUsername();
     }
 
+    /**
+     * Sets the last login credentials
+     * @param username The username used during the last login
+     * @param password The password used during the last login
+     */
     public void setLastLoginCredentials(String username, String password) {
-        global.setLastLoginCredentials(username, password);
+        sessionConfig.setLastLoginCredentials(username, password);
+        save();
     }
 
+    /**
+     * Retrieves a list of the last open filters
+     */
     public List<String> getLastOpenFilters() {
-        return global.getLastOpenFilters();
+        return sessionConfig.getLastOpenFilters();
     }
 
+    /**
+     * Retrieves a list of panel names
+     */
     public List<String> getPanelNames() {
-        return global.getPanelNames();
+        return sessionConfig.getPanelNames();
     }
-    
+
+    /**
+     * Retrieves a list of panel infos
+     */
     public List<PanelInfo> getPanelInfo() {
-        return global.getPanelInfo();
-    }
-    
-    public void setPanelInfo(List<PanelInfo> panelInfo) {
-        global.setPanelInfo(panelInfo);
+        return sessionConfig.getPanelInfo();
     }
 
     /**
-     * Interface to configuration files
+     * Sets the panel infos
+     * @param panelInfo The list of panel infos to set to
      */
+    public void setPanelInfo(List<PanelInfo> panelInfo) {
+        sessionConfig.setPanelInfo(panelInfo);
+        save();
+    }
 
     /**
-     * Boards
+     * Adds a board with the board name and its panels information
      */
-
     public void addBoard(String name, List<PanelInfo> panels) {
         assert name != null && panels != null;
-        global.addBoard(name, panels);
+        sessionConfig.addBoard(name, panels);
+        save();
     }
 
+    /**
+     * Retrieves a map of all the boards
```

Comment:
`Retrieves all the boards`. `map` is implementation specific and once we make board into a first-class citizen, the comment will become irrelevant.